afs/parsers.py

```diff
@@ -0,0 +1,211 @@
+with open(Path(__file__).parent.joinpath('template', 'jupyter_app_default_requirements.txt'), 'r') as f:
+    JUPYTER_APP_DEFAULT_REQUIREMENTS = f.read()
+
+with open(Path(__file__).parent.joinpath('template', 'jupyter_api_default_requirements.txt'), 'r') as f:
+    JUPYTER_API_DEFAULT_REQUIREMENTS = f.read()
+JUPYTER_APP_DEFAULT_COMMAND = 'bash'
+JUPYTER_API_DEFAULT_COMMAND = 'sh startup.sh'
+JUPYTER_API_STARTUP = """
+#!/bin/bash
+
+# Disable SQLite history
+mkdir -p /home/vcap/app/.ipython/profile_default
+cat << EOF > /home/vcap/app/.ipython/profile_default/ipython_kernel_config.py
+# Configuration file for ipython-kernel.
+c = get_config()
+c.HistoryManager.enabled = False
+EOF
+
+# Jupyter kernel gateway configs
+mkdir -p /home/vcap/app/.jupyter
+cat << EOF > /home/vcap/app/.jupyter/jupyter_kernel_gateway_config.py
+# Configuration file for jupyter-kernel-gateway.
+c.KernelGatewayApp.seed_uri='{notebook_name}'
+c.KernelGatewayApp.prespawn_count=5
+c.KernelGatewayApp.api='notebook-http'
+c.KernelGatewayApp.ip='0.0.0.0'
+c.KernelGatewayApp.port=$PORT
+c.NotebookHTTPPersonality.static_path='./static'
+EOF
+
+jupyter-kernelgateway
+"""
+
+
+def manifest_parser(notebook_path, output_dir=None, pypi_endpoint=None, manifest_yaml=False, afs_sdk_version=None):
+    """
+    The method parses the manifest in notebook, including  manifest.json, requirements.txt, runtime.txt, startup.sh.
+    :param str notebook_path: the path of notebook (.ipynb) will be parsed.
+    :param str output_dir: the files would be output in specific path
+    :param str pypi_endpoint: the requirement would be specific pypi server
+    :param bool manifest_yaml: write manifest.yml or not
+    :param str afs_sdk_version: parse manifest to specific afs sdk version requirement
+    :rtype: True or logger message
+    """
+    if not os.path.exists(notebook_path):
+        raise FileNotFoundError
+
+    if not notebook_path.endswith('.ipynb'):
+        raise ValueError('The file name extension is not ipynb.')
+
+    if output_dir is None:
+        output_dir = '.'
+
+    with open(notebook_path, 'r') as f:
+        notebook_content = json.loads(f.read())
+
+    manifest = next(
+        (notebook_content['cells'][0]['source']
+         for index, cell in enumerate(notebook_content['cells'])
+         if index == 0 and ''.join(cell['source']).replace('\n', '').startswith('manifest')), '')
+    manifest = ''.join(manifest).replace('\n', '')
+    manifest = config_to_dict(manifest, 'manifest') if manifest else {}
+    if 'error' in manifest:
+        message = {
+            'error': 'Save notebook {0} failed'.format(notebook_path),
+            'description': 'Parse manifest failed, Invalid format of manifest',
+            'status': 400
+        }
+        logger.error(message)
+        return message
+
+    message = {
+        'message': 'Save notebook {0} in progress'.format(notebook_path),
+        'description': 'Manifest: {0}'.format(ujson.dumps(manifest))
+    }
```

Comment:
In my original code, this `message` is used by `logger` to logging execution steps, Does SDK provide the same mechanism? If not, `message` should be removed.

Reply:
So far `logger` is not use now.
However, I want to add logger in modules next time.